coinflip/main.c

```diff
@@ -0,0 +1,394 @@
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/cm3/vector.h>
+#include <libopencm3/cm3/scb.h>
+#include <libopencm3/efm32/cmu.h>
+#include <libopencm3/efm32/gpio.h>
+#include <libopencm3/efm32/common/prs_common.h>
+#include <libopencm3/efm32/common/acmp_common.h>
+#include <libopencm3/efm32/timer.h>
+#include <libopencm3/efm32/wdog.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "captouch.h"
+#include "capsenseconfig.h"
+
+// Make this program compatible with Toboot-V2.0
+#include <toboot.h>
+TOBOOT_CONFIGURATION(TOBOOT_CONFIG_FLAG_AUTORUN);
+
+#define LED_GREEN_PORT GPIOA
+#define LED_GREEN_PIN GPIO0
+#define LED_RED_PORT GPIOB
+#define LED_RED_PIN GPIO7
+#define CAP0B_PORT GPIOE
+#define CAP0B_PIN GPIO12
+#define CAP1B_PORT GPIOE
+#define CAP1B_PIN GPIO13
+
+// Delay after the LED goes on before it turns off
+#define LED_ON_DELAY 1000
+
+// Delay after the LED goes off before the 'coin' can be flipped again
+#define LED_OFF_DELAY 2000
+
+// Minimum values for the capsense detect to work
+#define CAPSENSE_DETECT_MIN 20
+
+#pragma warning "Re-defining TIMER_CC_CTRL_INSEL because it's wrong"
+#undef TIMER_CC_CTRL_INSEL
+#define TIMER_CC_CTRL_INSEL (1 << 20)
+
+#define EFM_ASSERT(x)
+
+/**************************************************************************//**
+ * @brief This vector stores the latest read values from the ACMP
+ * @param ACMP_CHANNELS Vector of channels.
+ *****************************************************************************/
+static volatile uint32_t g_channel_values[4] = {0};
+
+/**************************************************************************//**
+ * @brief  This stores the maximum values seen by a channel
+ * @param ACMP_CHANNELS Vector of channels.
+ *****************************************************************************/
+static volatile uint32_t channelMaxValues[4] = {0};
+
+/** The current channel we are sensing. */
+static volatile uint8_t g_current_channel;
+
+/** Which generation of capsense we're on.  Monotonically increasing. */
+static volatile uint32_t g_capsense_generation;
+
+/** Set to true when we're freerunning capsense */
+static volatile bool g_capsense_running = false;
+
+/***************************************************************************//**
+ * @brief
+ *   Sets the ACMP channel used for capacative sensing.
+ *
+ * @note
+ *   A basic example of capacative sensing can be found in the STK BSP
+ *   (capsense demo).
+ *
+ * @param[in] acmp
+ *   Pointer to ACMP peripheral register block.
+ *
+ * @param[in] channel
+ *   The ACMP channel to use for capacative sensing (Possel).
+ ******************************************************************************/
+
+// Function prototypes so that our main loop can be at the top of the file for readability purposes
+static void ACMP_CapsenseChannelSet(uint32_t channel);
+static void CAPSENSE_Measure(uint32_t channel);
+static void CAPSENSE_Measure(uint32_t channel);
```

Comment:
Is this a duplicate of the previous line?

Reply:
Yes, removed one of these lines, good catch.